tests/test_geometry.py

```diff
@@ -51,3 +55,240 @@ def test_check_if_geometry_is_valid():
     mc.check_if_geometry_is_valid(1)
 
     mc.set_variable("Slot_Depth", save_slot_depth)
+
+
+def test_set_adaptive_parameter_value():
+    mc.set_adaptive_parameter_value("test_parameter", 100)
+
+
+def test_get_adaptive_parameter_value():
+    mc.set_adaptive_parameter_value("test_parameter_1", 100)
+
+    value = mc.get_adaptive_parameter_value("test_parameter_1")
+
+    assert value == 100
+
+
+def test_get_region():
+    region = mc.get_region("Stator")
+
+    assert region.name == "Stator"
+
+
+def test_set_region():
+    region = geometry.Region()
+    region.name = "testing_region"
+    region.colour = (0, 0, 255)
+    region.material = "Air"
+
+    mc.set_region(region)
+
+
+def test_get_entities_between_poly_start_end():
+    region = mc.get_region("Stator")
+    poly = [geometry.Line(region.entities[4].end, region.entities[7].end)]
+
+    entities_to_remove, start_index = mc.get_entities_between_poly_start_end(region, poly)
+
+    assert entities_to_remove == region.entities[4:7:1]
+    assert start_index == 4
+
+
+def test_save_adaptive_script():
+    mc.save_adaptive_script(get_dir_path() + r"\test_files\adaptive_templates_script.py")
+
+
+def test_region_add_entity():
+    region = geometry.Region()
+    region.add_entity(geometry.Line((0, 0), (1, 1)))
+
+    assert len(region.entities) == 1
```

Comment:
Can you assert anything about the object you've added. Would also be good to get the arc case checked